nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/test/java/org/apache/nifi/json/TestJsonTreeRowRecordReader.java

```diff
@@ -296,11 +298,34 @@ void testReadOneLinePerJSON() throws IOException, MalformedRecordException {
 
     @Test
     void testReadMultilineJSON() throws IOException, MalformedRecordException {
+        testReadAccountJson("src/test/resources/json/bank-account-multiline.json", false, null);
+    }
+
+    @Test
+    void testReadJSONStringTooLong() {
+        final StreamConstraintsException mre = assertThrows(StreamConstraintsException.class, () ->
+                testReadAccountJson("src/test/resources/json/bank-account-multiline.json", false, StreamReadConstraints.builder().maxStringLength(2).build()));
+        assertEquals("String length (8) exceeds the maximum length (2)", mre.getMessage());
+    }
+
+    @Test
+    void testReadJSONComments() throws IOException, MalformedRecordException {
+        testReadAccountJson("src/test/resources/json/bank-account-comments.jsonc", true, StreamReadConstraints.builder().maxStringLength(20_000).build());
+    }
+
+    @Test
+    void testReadJSONDisallowComments() {
+        final MalformedRecordException mre = assertThrows(MalformedRecordException.class, () ->
+            testReadAccountJson("src/test/resources/json/bank-account-comments.jsonc", false, StreamReadConstraints.builder().maxStringLength(20_000).build()));
+        assertEquals("Could not parse data as JSON", mre.getMessage());
```

Comment:
This assertion can be removed, or perhaps adjusted to look for the word `parse` by itself.